python/paddle/fluid/framework.py

```diff
@@ -284,7 +284,17 @@ def _current_expected_place():
     global _global_expected_place_
     if _global_expected_place_ is None:
         if core.is_compiled_with_cuda():
-            _global_expected_place_ = core.CUDAPlace(0)
+            try:
+                device_count = core.get_cuda_device_count()
+            except Exception as e:
+                device_count = 0
```

Comment:
I think we need to do something if an Exception happens instead of just ignore it, we can raise warnings at least. It can be refined in the next PR.